### PR TITLE
Added primary modal for all future mainpage modal work, and added search guidelines.

### PR DESCRIPTION
**Pull request by jesperlauridsen**
Added a primaryModal window for all to SolrWayback
Added SearchGudelines
Added service to get search help guide
Reworked styling og search guide

There might be some CSS conflicts/changes that needs to be worked in when SOBACK-23-wordcloud has been merged into newfrontend.

### CODE REVIEW

File: src/js/src/components/SearchBox.vue [hunk 1]
```diff
@@ -66,14 +69,13 @@ import SearchUploadFile from './SearchUploadFile.vue'
 export default {
   components: {
     AppliedSearchFacets,
-    SearchUploadFile
+    SearchUploadFile,
```
Comment (jorntx) [inline src/js/src/components/SearchBox.vue:72]: Unneeded trailing comma

File: src/js/src/components/SearchBox.vue [hunk 2]
```diff
@@ -66,14 +69,13 @@ import SearchUploadFile from './SearchUploadFile.vue'
...
   },
   mixins: [HistoryRoutingUtils, SearchUtils],
   data () {
     return {    
       futureQuery:'',
-      showUploadFileSearch: false
-
+      showUploadFileSearch: false,
```
Comment (jorntx) [inline src/js/src/components/SearchBox.vue:78]: Unneeded trailing comma

File: src/js/src/components/modalComponents/PrimaryModal.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    SearchGuidelines
+  },
+  data() {
+    return {
+    }
+  },
+  computed: {
+    ...mapState({
+      showModal: state => state.Modal.showModal,
+      currentModal: state => state.Modal.currentModal,
```
Comment (jorntx) [inline src/js/src/components/modalComponents/PrimaryModal.vue:29]: Good choice with separate store for this 👍

File: src/js/src/components/modalComponents/PrimaryModal.vue [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+export default {
+  name: 'PrimaryModal',
+  components: {
+    SearchGuidelines
+  },
+  data() {
+    return {
+    }
+  },
```
Comment (jorntx) [inline src/js/src/components/modalComponents/PrimaryModal.vue:25]: Unused code

File: src/js/src/components/modalComponents/SearchGuidelines.vue [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+</template>
+
+<script>
+
+import { requestService } from '../../services/RequestService'
+
+export default {
+  name: 'SearchGuidelines',
+  data: () => ({
+        searchGuidelines:'',
```
Comment (jorntx) [inline src/js/src/components/modalComponents/SearchGuidelines.vue:14]: Unneeded trailing comma

File: src/js/src/store/modules/modal.store.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+// Global modal state
+
+import Vue from 'vue'
```
Comment (jorntx) [inline src/js/src/store/modules/modal.store.js:3]: Unused import statement

File: src/js/src/views/SolrWayback.vue [hunk 7]
```diff
@@ -33,17 +35,19 @@ export default {
...
    Notifications,
    LoadingOverlay,
-   AboutComponent
+   AboutComponent,
+   PrimaryModal
   },
   mixins: [SearchUtils],
   data: () => ({
-        scrolledFromTop:false
+        scrolledFromTop:false,
```
Comment (jorntx) [inline src/js/src/views/SolrWayback.vue:43]: Unneeded trailing comma

File: src/js/src/views/SolrWayback.vue [hunk 8]
```diff
@@ -33,17 +35,19 @@ export default {
...
   }),
   computed: {
     ...mapState({
       searchAppliedFacets: state => state.Search.searchAppliedFacets,
       query: state => state.Search.query,
-      solrSettings: state => state.Search.solrSettings
+      solrSettings: state => state.Search.solrSettings,
+      showModal: state => state.Modal.showModal,
```
Comment (jorntx) [inline src/js/src/views/SolrWayback.vue:50]: Unneeded trailing comma

File: src/js/src/views/SolrWayback.vue [hunk 9]
```diff
@@ -74,7 +78,7 @@ export default {
        to.query.urlSearch !== from.query.urlSearch ||
        to.query.grouping !== from.query.grouping ||
        to.query.facets !== from.query.facets)
-    }
+    },
```
Comment (jorntx) [inline src/js/src/views/SolrWayback.vue:81]: Unneeded trailing comma

File: src/js/vue.config.js [hunk 10]
```diff
@@ -77,6 +77,11 @@ module.exports = {
         pathRewrite: { '^/services/frontend/image/pagepreview/': '' },
         changeOrigin: true
       },
+    '^/services/frontend/help/search/': {
+        target: 'http://localhost:8080/solrwayback/services/frontend/help/search/',
+        pathRewrite: { '^/services/frontend/help/search/': '' },
+        changeOrigin: true
+      },
```
Comment (jorntx) [inline src/js/vue.config.js:84]: Unneeded trailing comma

**Review by jorntx — APPROVED** [review]
Looks fine. I get a 500 service error when trying to view help text locally. I suspect it is some kind of Tomcat issue in my setup.

I only have some minor cleanup comments.

**Comment by jorntx** [comment]
@jesperlauridsen fix merge conflicts - thanks in advance. You can address the clean up comments at the same time as well but the pull request is approved as it is.

**Comment by jesperlauridsen** [comment]
Merged newfrontend and fixed errors. Should be good to go now, @jorntx

**Pull request merged**